UnityProject/Assets/Scripts/Objects/Research/Artifacts/Artifact.cs

```diff
@@ -9,182 +18,484 @@ public class ArtifactSprite
+						break;
+				}
+			}
+
+			//Randomises ID and adds it to name, this is done for two reasons:
+			//To prevent duplication exploits, server knows if youve researched an artifact before and won't give you extra RP or credits for repeated research.
+			//So players know which artifact is which if they have the same sprites
+			//ID is in form: A000 - Z999
+			bool generated = false;
+
+			while (iDList.Contains(ID) == true || generated == false)
+			{
+				int num = Random.Range(0, 26);
+				ID = $"{(char)('a' + num)}{Random.Range(0, 1000).ToString("000")}";
+				ID = ID.ToUpperInvariant();
+				generated = true;
+			}
+
+			iDList.Add(ID);
+
+			GetComponent<ObjectAttributes>().ServerSetArticleName("Artifact - " + ID);
+
+			ArtifactClass chosenClass;	
+
+			//Randomises the effects of the artifacts, probabilities of certain effects change with composition
+			if (AreaEffect == null)
+			{
+				chosenClass = PickClass();
+				AreaEffect = ArtifactDataSO.AreaEffects[(int)chosenClass].AreaArtifactEffectList.PickRandom();
+			}
+			if (InteractEffect == null)
+			{
+				chosenClass = PickClass();
+				InteractEffect = ArtifactDataSO.InteractEffects[(int)chosenClass].InteractArtifactEffectList.PickRandom();
+			}
+			if (DamageEffect == null)
+			{
+				chosenClass = PickClass();
+				DamageEffect = ArtifactDataSO.DamageEffect[(int)chosenClass].DamageArtifactEffectList.PickRandom();
+			}
+
+			artifactData.AreaEffectValue = AreaEffect.GuessIndex;
+			artifactData.InteractEffectValue = InteractEffect.GuessIndex;
+			artifactData.DamageEffectValue = DamageEffect.GuessIndex;
+
+			artifactData.ID = ID;
+
+			ArtifactData temp = artifactData;
+			artifactData = new ArtifactData();
+			artifactData = temp;
+
+			ToggleDormancy(isDormant); //Doesn't change dormant value but updates sprite
+
+			//Initalises Radiation for artifacts with uranium composition.
+			if (artifactData.radiationlevel > 0)
+			{
+				radiationProducer.NetSetActive(true);
+				radiationProducer.SetLevel(artifactData.radiationlevel);
+			}
+
+			UpdateManager.Add(UpdateMe, AreaEffect.coolDown);
 		}
 
-		// check if artifact is always active and just emits aura
-		if (currentTrigger == ArtifactTrigger.ALWAYS_ACTIVE)
+		public void OnDespawnServer(DespawnInfo info)
 		{
-			TryActivateAura();
+			integrity.OnApplyDamage.RemoveListener(DoDamageEffect);
+
+			// remove it from global artifacts registry
+			if (ServerSpawnedArtifacts.Contains(this))
+				ServerSpawnedArtifacts.Remove(this);
+
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, UpdateMe);
 		}
-	}
 
-	public void OnSpawnServer(SpawnInfo info)
-	{
-		// select trigger for artifact
-		if (!ForceTrigger)
+		private void OnDestroy()
 		{
-			ServerSelectRandomTrigger();
+			// remove it from global artifacts registry
+			if (ServerSpawnedArtifacts.Contains(this))
+				ServerSpawnedArtifacts.Remove(this);
 		}
-		else
+
+
+		#endregion
+
+		public static void ClearStatics()
```

Comment:
Think you need this too (CC @ThatDan123):

```suggestion
		[RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
		public static void ClearStatics()
```